buildSrc/src/main/groovy/io/micronaut/guides/IndexGenerator.groovy

```diff
@@ -66,12 +76,30 @@ class IndexGenerator {
             text = text.replace("@breadcrumb@", breadcrumb)
         }
         text = text.replace("@title@", title)
+        String twittercard = ''
+        if (singleGuide) {
+            twittercard = twitterCardHtml(Paths.get(buildDir.absolutePath, "/dist").toFile(), metadatas.get(0))
+        }
+        text = text.replace("@twittercard@", twittercard)
         text = text.replace("@bodyclass@", 'guideindex')
         text = text.replace("@toccontent@", '')
         text = text.replace("@content@", index)
         text
     }
 
+    static String twitterCardHtml(File distDir, GuideMetadata guideMetadata) {
+        String rootUrl = System.getenv("CI") ? GUIDES_URL : ""
+        String filename = (Paths.get(distDir.absolutePath, "/images/cards/" + guideMetadata.slug + ".png").toFile().exists()) ?
+                guideMetadata.slug + ".png" : DEFAULT_CARD
+"""\
+<meta name='twitter:card' content='summary_large_image'/>
+<meta name='twitter:image' content='${rootUrl}/images/cards/${filename}'>
+<meta name="twitter:creator" content="${TWITTER_MICRONAUT}">
+<meta name="twitter:site" content="${TWITTER_MICRONAUT}">
+<meta name='twitter:title' content='${guideMetadata.title ?: DEFAULT_TITLE}'/>
+<meta name='twitter:description' content='${guideMetadata.intro ?: DEFAULT_INTRO}'/>"""
```

Comment:
`twitter:image`, `twitter:creator` and `twitter:title` tags are not closed properly. You put `>` instead of `/>`